tests/Components/Jobs/UpdateProfileTest.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Tests\Components\Jobs;
+
+use Tests\TestCase;
+use App\Jobs\UpdateProfile;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+
+class UpdateProfileTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    /** @test */
+    public function we_can_update_a_user_profile()
+    {
+        $user = $this->login(['name' => 'John Doe', 'email' => 'john@example.com', 'username' => 'johndoe']);
+
+        (new UpdateProfile($user, ['bio' => 'my bio', 'name' => 'John Doe Junior']))->handle();
+
+        $this->assertEquals('my bio', $user->bio());
+        $this->assertEquals('John Doe Junior', $user->name());
+    }
```

Comment:
It's not needed to login the user. Since you're testing a component no "state" of the active user is needed. You can re-write this as follow:

```php
    /** @test */
    public function we_can_update_a_user_profile()
    {
        $user = (new UpdateProfile($this->createUser(), ['bio' => 'my bio', 'name' => 'John Doe Junior']))->handle();

        $this->assertEquals('my bio', $user->bio());
        $this->assertEquals('John Doe Junior', $user->name());
    }
```

Make sure that the `handle` method returns the updated user.